### PR TITLE
migrate x-pack/osquerybeat pipeline

### PR DESCRIPTION
**Pull request by sharbuz**
## What is the problem this PR solves?

Jenkins->Buildkite pipelines migration 

Migrate `x-pack osquerybeat` pipeline

Example of the pipeline:

https://buildkite.com/elastic/beats-xpack-osquerybeat/builds/292

## Related issues
https://github.com/elastic/ingest-dev/issues/1693

### CODE REVIEW

File: .buildkite/scripts/generate_xpack_osquerybeat_pipeline.sh [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+              - "${IMAGE_WIN_2019}"
+        artifact_paths: "${BEATS_PROJECT_NAME}/build/*.*"
+
+YAML
+else
+  echo "The conditions don't match to requirements for generating pipeline steps."
+  exit 0
+fi
+
+if are_conditions_met_macos_tests; then
```
Comment (v1v) [inline .buildkite/scripts/generate_xpack_osquerybeat_pipeline.sh:81]: What's the reason MacOS M1 is not here?

See https://github.com/elastic/beats/blob/12ae03f48267f7af774b376d92168960c2129607/x-pack/osquerybeat/Jenkinsfile.yml#L35-L47

**Review by v1v — COMMENTED** [review]
LGTM with just a comment regarding MacOS M1 support

File: .buildkite/scripts/generate_xpack_osquerybeat_pipeline.sh (hunk 1 above)
Comment (sharbuz) [reply .buildkite/scripts/generate_xpack_osquerybeat_pipeline.sh:81]: We haven't had the agent for MacOS M1 yet.

**Review by v1v — APPROVED** [review]

**Review by dliappis — APPROVED** [review]
LGTM

**Comment by sharbuz** [comment]
run docs-build

**Pull request merged**